opentelemetry-sdk/src/logs/record.rs

```diff
@@ -97,6 +97,27 @@ impl opentelemetry::logs::LogRecord for LogRecord {
     }
 }
 
+impl LogRecord {
+    /// Provides an iterator over the attributes in the `LogRecord`.
+    pub fn attributes_iter(&self) -> impl Iterator<Item = &(Key, AnyValue)> {
+        self.attributes
+            .as_ref()
+            .map_or_else(|| [].iter(), |attrs| attrs.iter())
+    }
+
+    /// Returns the number of attributes in the `LogRecord`.
+    pub fn attributes_len(&self) -> usize {
+        self.attributes.as_ref().map_or(0, |attrs| attrs.len())
+    }
+
+    /// Returns true if the `LogRecord` contains the specified attribute.
```

Comment:
I think this is not required to be public. If an exporter needs it, then can write a helper function to get this by using the above 2 methods.

Reply:
I think we need to offer API to modify an existing entry, remove an existing entry too. These are critical to ensure we allow redaction, enrichment capabilities. Lets track it as separate issue for adding in the future.